src/config/schema.rs

```diff
@@ -126,10 +133,52 @@ pub struct S3 {
     pub cache_properties: Option<ObjectCacheProperties>,
 }
 
+impl S3 {
+    pub fn from_bucket_and_options(
+        bucket: String,
+        map: &HashMap<String, String>,
+    ) -> Result<Self, DataFusionError> {
+        Ok(S3 {
+            region: map.get("region").cloned(),
+            access_key_id: map
+                .get("access_key_id")
+                .ok_or(DataFusionError::Execution(
+                    "'access_key_id' not found in provided options".to_string(),
+                ))?
+                .clone(),
+            secret_access_key: map
+                .get("secret_access_key")
+                .ok_or(DataFusionError::Execution(
+                    "'secret_access_key' not found in provided options".to_string(),
+                ))?
+                .clone(),
+            endpoint: map.get("endpoint").cloned(),
+            bucket,
+            cache_properties: Some(ObjectCacheProperties::default()),
```

Comment:
Maybe we should be setting this to `None` in this case)

Reply:
Is there a difference between `None` and `Some(ObjectCacheProperties::default())` (if someone doesn't want their object store contents to be cached, can they represent that in the config)? If not, we could replace the type with `ObjectCacheProperties` instead of the optional.

Reply:
> if someone doesn't want their object store contents to be cached, can they represent that in the config

this is currently represented by not specifying the `[object_store.cache_properties]` section in the config, and I think it's wise to leave it like that (i.e. keep it optional).

I'm just wondering whether it's worth it to setup the temp file caching machinery for a external table, and I'm leaning towards no for starters.

Reply:
It could be useful. The reason is that we originally set up caching for the HTTP object store because DataFusion would call our object store requesting very small byte ranges very frequently, so we'd get slowed down by the latency (this could now have been fixed, but could still be a problem for S3/GCS).